.github/file-filters.yml

```diff
@@ -1,44 +1,73 @@
 # This is used by the action https://github.com/dorny/paths-filter
 
-# TODO: There are some meta files that we potentially could ignore for both front/backend,
-# as well as some configuration files that should trigger both
-frontend_components_lintable: &frontend_components_lintable
-  - '**/*.[tj]{s,sx}'
-
-frontend_lintable: &frontend_lintable
-  - *frontend_components_lintable
-  - '**/tsconfig*.json'
-  - '{package,now,vercel}.json'
-
-yarn_lockfile: &yarn_lockfile
-  - 'yarn.lock'
-
-eslint_config: &eslint_config
-  - '.eslint*'
-
-# `frontend_src` filters on files that are frontend changes excluding
-# changes to the tests/ directory.
-# If you want to filter on *all* frontend files, use the `frontend_all` filter.
-frontend_src: &frontend_src
-  - *yarn_lockfile
-  - *eslint_config
-  - '!(tests)/**/*.[tj]{s,sx}'
-  - '**/tsconfig*.json'
-  - '{package,now,vercel}.json'
-  - '**/*.less'
-  - 'static/**'
-  - 'fixtures/search-syntax/**/*'
-  - '.github/workflows/frontend.yml'
-
+sentry_specific_workflow: &sentry_specific_workflow
+  - added|modified: '.github/workflows/frontend.yml'
+sentry_specific_test_files: &sentry_specific_test_files
+  - added|modified: 'tests/js/**/*'
+  - added|deleted|modified: 'fixtures/search-syntax/**/*'
+  - added|deleted|modified: 'fixtures/js-stubs/**/*'
+
+# Provides list of changed app files to lint against (stylelint)
+lintable_css_in_js_modified: &lintable_css_in_js_modified
+  - added|modified: '**/*.[tj]{s,sx}'
+
+# Trigger for when we must run full lint (stylelint)
+lintable_css_in_js_modified_rules_changed: &lintable_css_in_js_modified_rules_changed
+  - *sentry_specific_workflow
+  - added|modified: '.github/file-filters.yml'
+  - added|modified: 'yarn.lock'
+  - added|deleted|modified: 'stylelint.*'
+  - *lintable_css_in_js_modified
+
+# Provides list of changed files to lint against (eslint)
+lintable_modified: &lintable_modified
+  - added|modified: '**/*.[tj]{s,sx}'
+  - added|modified: '**/*.{html,less,json,yml,md}'
+
+# Trigger for when we must run full lint (eslint)
+lintable_modified_rules_changed: &lintable_modified_rules_changed
+  - *sentry_specific_workflow
+  - added|modified: '.github/file-filters.yml'
+  - added|modified: 'package.json'
+  - added|modified: 'yarn.lock'
+  - added|deleted|modified: '.prettierignore'
+  - added|deleted|modified: '.eslintignore'
+  - added|deleted|modified: '.eslint*'
+  - *lintable_modified
```

Comment:
Actually, doesn't this mean that we will always be linting all of the files?

Originally the idea was that we would only lint+fix changed files because of the strict lint ruleset that runs on PRs (and in precommit) vs a relaxed ruleset run across all files in master. This was because of our iterative migration process where we wanted to ensure that the code you touch would be updated to the stricter ruleset.

If we're getting rid of the strict ruleset, then there's no need to lint/fix only changed files if we're linting all of the files.

Reply:
ah yeah, i messed it up again. I think i need to split out the 'rules changed' condition from the 'files or rules changed' condition. 

If we're ok with linting speed in master, then we could just dump this changed-files only business and lint everything every time. it'll be simpler, but should be slower. On my M1 laptop it takes 90 seconds to run `yarn lint` which is all the lint steps. On github it looks a bit slower, more like 150seconds without stylelint, and `tsc` is another 60seconds. So more like 4-5 mins total. Tests run in parallel with that, taking ~5mins as well.